lib/cmock_generator.rb

```diff
@@ -206,6 +207,9 @@ def create_mock_implementation(file, function)
     args_string += (", " + function[:var_arg]) unless (function[:var_arg].nil?)
 
     # Create mock function
+    if (@weak)
+        file << "#{function_mod_and_rettype} #{function[:name]}(#{args_string}) __attribute ((weak));\n"
```

Comment:
This is really compiler specific. How about you add an argument to the weak (instead of true-false) what kind of attribute needs to be added? I think IAR has a weak prefix which might complicate things.